### PR TITLE
Tables, chairs and the vikings are moving to the tables

### LINKED ISSUE #38: Allow guiding vikings to tables
**Description**
The player should be able to tell the vikings where to sit. The vikings should then navigate (or teleport for now) to a chair at that table.

**Subtasks**
- [x] Viking should have a "waiting in line" state
- [x] Can interact with viking to tell them where to sit
- [x] Viking can navigate to the table

### LINKED ISSUE #39: Tables and chairs
**Description**
We need to have a way to represent tables and chairs.

**Subtasks**
- [x] Tables should be interactable (used for telling vikings what table to sit at)
- [x] Tables should keep track of its chairs and which are occupied

### PR DESCRIPTION
**Pull request by smeas**
Closes #38, Closes #39 

**Description**  
Implemented chairs, tables and allow the player to guide vikings to tables.

Vikings use nav mesh to navigate. So make sure to have one set up in the scene for it to work.

**List of changes**  
- Added chairs and tables
- Made the player able to guide the vikings to a table
- Made the InteractionDetector not require things to have rigidbodies
- Adjusted pivots for viking and player
- Extended viking state machine with more core state methods (Enter & Exit)

**Additional context**  

https://user-images.githubusercontent.com/8546813/115030557-84e1c080-9ec7-11eb-924a-14285949d8cb.mp4

### CODE REVIEW

File: Assets/00_Content/Scripts/Interactables/Interactable.cs [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 	public class Interactable : MonoBehaviour {
 
 		public virtual void CancelInteraction() {
 		}
-		public virtual void Interact() {
+		public virtual void Interact(GameObject player, PickUp item) {
+		}
+
+		public virtual bool CanInteract(GameObject player, PickUp item) {
+			return true;
```
Comment (tzaarela) [inline Assets/00_Content/Scripts/Interactables/Interactable.cs:12]: I dont see a reason to force an interaction to require an item. I can see examples where we want to interact with things not using an item. I would just use a method overload for when an item is required.

**Review by tzaarela — CHANGES_REQUESTED** [review]
Looks great! Just a minor thing I would like to see changed.

File: Assets/00_Content/Scripts/Interactables/Interactable.cs (hunk 1 above)
Comment (smeas) [reply Assets/00_Content/Scripts/Interactables/Interactable.cs:12]: The idea is that the player always sends along whatever it is holding, and then let the interactable decide if such an interaction is possible. The interactable can choose to ignore the item if it doesn't care.

I think having an overload would just complicate the usage. You would have to implement both overloads if you want to cover both cases. I believe that just always passing along the held item makes the most sense.

Comment (tzaarela) [reply Assets/00_Content/Scripts/Interactables/Interactable.cs:12]: So we just send NULL when interacting with things not requireing an item or if we don´t hold an item.

**Review by tzaarela — APPROVED** [review]

**Pull request merged**